x/gamm/pool-models/balancer/util_test.go

```diff
@@ -36,3 +38,51 @@ func createTestContext(t *testing.T) sdk.Context {
 
 	return sdk.NewContext(ms, tmtypes.Header{}, false, logger)
 }
+
+func assertExpectedSharesErrRatio(t *testing.T, expectedShares, actualShares sdk.Int) {
+	allowedErrRatioDec, err := sdk.NewDecFromStr(allowedErrRatio)
+	require.NoError(t, err)
+
+	errTolerance := osmoutils.ErrTolerance{
+		MultiplicativeTolerance: allowedErrRatioDec,
+	}
+
+	require.Equal(
+		t,
+		0,
+		errTolerance.Compare(expectedShares, actualShares),
+		fmt.Sprintf("expectedShares: %s, actualShares: %s", expectedShares.String(), actualShares.String()))
+}
+
+func assertExpectedLiquidity(t *testing.T, tokensJoined, liquidity sdk.Coins) {
+	require.Equal(t, tokensJoined, liquidity)
+}
+
+// assertPoolStateNotModified asserts that sut (system under test) does not modify
+// pool state.
+func assertPoolStateNotModified(t *testing.T, pool *balancer.Pool, sut func()) {
+	// We need to make sure that this method does not mutate state.
+	oldPoolAssets := pool.GetAllPoolAssets()
+	oldLiquidity := pool.GetTotalPoolLiquidity(sdk.Context{})
+	oldShares := pool.GetTotalShares()
+
+	sut()
+
+	newPoolAssets := pool.GetAllPoolAssets()
+	newLiquidity := pool.GetTotalPoolLiquidity(sdk.Context{})
+	newShares := pool.GetTotalShares()
+
+	require.Equal(t, oldPoolAssets, newPoolAssets)
+	require.Equal(t, oldLiquidity, newLiquidity)
+	require.Equal(t, oldShares, newShares)
+}
+
+// assertPanic if expectPanic is true, asserts that sut (system under test)
+// panics. If expectPanic is false, asserts that sut does not panic.
+func assertPanic(t *testing.T, expectPanic bool, sut func()) {
```

Comment:
what's `sut` supposed to stand for? Maybe just call this `fn`?

Reply:
It means system under test, aI translated this in comment as well.

Imo this is a common name to refer to something that is being tested: https://en.wikipedia.org/wiki/System_under_test

Reply:
If people still prefer `fn`, happy to change

Reply:
interesting haha, ok!

Reply:
had the same question 😃

Reply:
I've taken another look at renaming to `fn`. I think it would be too general since `fn` can be anything and `sut` makes it more precise that we are running the "function/system" under test by calling `sut()`

Reply:
Going to merge. If anyone disagrees with my latest comment, please let me know and I will change it in a follow-up PR